### PR TITLE
[REVIEW] make `var/std` unsupported only at debug build

### PR DESCRIPTION
**Pull request by kovaltan**
Due to cuda compiler bug, redutctoin `var/std` cause compilation failure only at debug build. The bug will be fixed at cuda 10.2.
This PR is a workaround against the compilation failure, this PR make `var/std` unsuppoted only at debug build.

Related PR: #1993 #2066

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +61,7 @@
 - PR #2223 Fix CUDA invalid configuration errors reported after loading small compressed ORC files
 - PR #2162 Setting is_unique and is_monotonic-related attributes
 - PR #2244 Fix ORC RLEv2 delta mode decoding with nonzero residual delta width
+- PR #2297 Work around `var/std` unsuppoted only at debug build
```
Comment (devavret) [inline CHANGELOG.md]:
```suggestion
- PR #2297 Work around `var/std` unsupported only at debug build
```

File: cpp/src/reductions/std.cu [hunk 2]
```diff
@@ -24,8 +24,14 @@
 
 gdf_scalar cudf::reduction::standard_deviation(gdf_column const& col, gdf_dtype const output_dtype, gdf_size_type ddof, cudaStream_t stream)
 {
+#if !defined(__CUDACC_DEBUG__)
```
Comment (devavret) [inline cpp/src/reductions/std.cu]: Is it also possible to add CUDA version to the check, considering we'll be fixing it in later CUDA versions. If you're unsure which version it'll be, you can add a TODO.

File: CHANGELOG.md (hunk 1 above)
Comment (kovaltan) [reply CHANGELOG.md]: Fixed, thanks,

File: cpp/src/reductions/std.cu (hunk 2 above)
Comment (kovaltan) [reply cpp/src/reductions/std.cu]: Added "TODO" text

**Comment by jrhemstad** [comment]
@devavret can you confirm this fixes the issue you were seeing?

**Comment by randerzander** [comment]
fwiw I've had the same build failures for the last day or so. Turning off debug mode let the build proceed.

**Pull request merged**